components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/AuthorizationHandlerManager.java

```
@@ -121,6 +121,55 @@ public OAuth2AuthorizeRespDTO handleAuthorization(OAuth2AuthorizeReqDTO authzReq
         return authorizeRespDTO;
     }
 
+    public OAuthAuthzReqMessageContext handleScopeValidation(OAuth2AuthorizeReqDTO authzReqDTO)
+            throws IdentityOAuth2Exception, InvalidOAuthClientException {
+
+        OAuthAuthzReqMessageContext authzReqMsgCtx = getOAuthAuthzReqMessageContext(authzReqDTO);
+        ResponseTypeHandler authzHandler = getResponseHandler(authzReqDTO);
+        validateAuthzRequestScopes(authzReqMsgCtx, authzHandler);
```

This method should throw an exception if validation of request scopes fails